assets/js/base/components/product-list/index.js

```diff
@@ -175,7 +179,9 @@ const ProductList = ( {
 	const { contentVisibility } = attributes;
 	const perPage = attributes.columns * attributes.rows;
 	const totalPages =
-		! Number.isFinite( totalProducts ) && isPreviousTotalQueryEqual
+		! Number.isFinite( totalProducts ) &&
+		Number.isFinite( previousQueryTotals?.totalProducts ) &&
+		isEqual( totalQuery, previousQueryTotals?.totalQuery )
```

Comment:
The check that was named `isPreviousTotalQueryEqual`, has been moved here. Instead of checking that `previousQueryTotals` is an object, we directly check if it has the `totalProducts` property and it's a finite number.